### PR TITLE
 rtl8852au: unstable-2024-04-16 -> unstable-2024-05-06

### PR DESCRIPTION
**Pull request by lonyelon**
## Description of changes

https://github.com/lwfinger/rtl8852au/commit/865ab0fa91471d595c283d2f3db323f7f15455f5

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
no new breakages compared to hydra

File: pkgs/os-specific/linux/rtl8852au/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   pname = "rtl8852au";
-  version = "${kernel.version}-unstable-2024-04-16";
+  version = "${kernel.version}-unstable-2024-05-06";
 
   src = fetchFromGitHub {
     owner = "lwfinger";
-    repo = "rtl8852au";
-    rev = "5894bc6fed2bcaa525d13fcee1edada8aba67f2b";
-    hash = "sha256-R4Yb/jbh3nMgM41ByFjtkCMbsh/mmMRJ7CcvCRUvKu8=";
+    repo = pname;
```
Comment (pbsds) [inline pkgs/os-specific/linux/rtl8852au/default.nix:9]:
```suggestion
    repo = "rtl8852au";
```

please undo this, this is considered an anti pattern

Comment (JohnRTitor) [reply pkgs/os-specific/linux/rtl8852au/default.nix]: Done.

**Review by JohnRTitor — APPROVED** [review]
LGTM though, I would consider moving to a different upstream repo, perhaps a fork, at some point, since Larry has passed away.

**Comment by lonyelon** [comment]
I did not know, it was weird that he did not keep uploading commits. I can try to take these drivers myself, but IDK if I would be able to be actively maintaining them.

**Comment by lonyelon** [comment]
RIP Larry, he truly was great, keeping so many wireless drivers out of pure passion.

**Pull request merged**